src/chocolatey/infrastructure.app/commands/ChocolateyExportCommand.cs

```diff
@@ -0,0 +1,185 @@
+                     option => configuration.ExportCommand.IncludeVersionNumbers = option != null)
+                ;
+        }
+
+        public void handle_additional_argument_parsing(IList<string> unparsedArguments, ChocolateyConfiguration configuration)
+        {
+            if (unparsedArguments.Count > 0)
+            {
+                throw new ApplicationException("Please see the help menu for the export command");
+            }
+        }
+
+        public void handle_validation(ChocolateyConfiguration configuration)
+        {
+            // Currently, no additional validation is required.
+        }
+
+        public void help_message(ChocolateyConfiguration configuration)
+        {
+            this.Log().Info(ChocolateyLoggers.Important, "Export Command");
+            this.Log().Info(@"
+Export all currently installed packages to a file.
+
+This is especially helpful when re-building a machine that was created
+using Chocolatey.  Export all packages to a file, and then re-install
+those packages onto new machine using `choco install packages.config`.
+");
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Usage");
+            "chocolatey".Log().Info(@"
+    choco export [<options/switches>]
+");
+
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Examples");
+            "chocolatey".Log().Info(@"
+    choco export
+    choco export --include-version-numbers
+    choco export --output-file-path=""c:\temp\packages.config""
+    choco export --output-file-path=""c:\temp\packages.config"" --include-version-numbers
+
+NOTE: See scripting in the command reference (`choco -?`) for how to
+ write proper scripts and integrations.
+
+");
+
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Exit Codes");
+            "chocolatey".Log().Info(@"
+Exit codes that normally result from running this command.
+
+Normal:
+ - 0: operation was successful, no issues detected
+ - -1 or 1: an error has occurred
+
+If you find other exit codes that we have not yet documented, please
+ file a ticket so we can document it at
+ https://github.com/chocolatey/choco/issues/new/choose.
+
+");
+
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Options and Switches");
+        }
+
+        public bool may_require_admin_access()
+        {
+            return false;
+        }
+
+        public void noop(ChocolateyConfiguration configuration)
+        {
+            this.Log().Info("Export would have been with options: {0} Output File Path={1}{0} Include Version Numbers:{2}".format_with(Environment.NewLine, configuration.ExportCommand.OutputFilePath, configuration.ExportCommand.IncludeVersionNumbers));
+        }
+
+        public void run(ChocolateyConfiguration configuration)
+        {
+            var packageResults = _nugetService.get_all_installed_packages(configuration);
+            var settings = new XmlWriterSettings { Indent = true, Encoding = new UTF8Encoding(false) };

```

Comment:
Would like to be able to include other items to build out a bigger packages.config (for future use). One thing I would like to see here at the very least is to refactor to IXmlService and XmlService.

Reply:
@ferventcoder do you remember what the thought process was here again?  Looking at the code again just now, and I am not immediately clear on why this would help here.  Thanks

Reply:
I have went through some notes that I had on this, as well as watching back the videos of the Twitch streams that we did on this, and the issue here was (I believe) that if we go down this route, there was no direct way of including/excluding specific attributes within the resulting XML file.  Not without implementing a custom XML Serializer/DeSerializer for the class.  I would suggest that this might be something that we could look at if/when we bring in the ability to include additional attributes into the resulting XML file, but for now, I think the code that we have here is a good compromise.